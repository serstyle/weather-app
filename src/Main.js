import React, { Component } from 'react';
import { connect } from 'react-redux'
import Home from './components/Home'
import Search from './components/Search'
import {handleChangeCity, fetchWeather} from './actions.js'


const mapStateToProps = (state) =>{
  return{
    city: state.handleChangeCity.city,
    currentTemp: state.fetchWeather.currentTemp,
    currentWeath: state.fetchWeather.currentWeath,
    currentIcon: state.fetchWeather.currentIcon,
    currentCity: state.fetchWeather.currentCity,
    isPending: state.fetchWeather.isPending,
    error: state.fetchWeather.error
}}

const mapDispatchToProps = (dispatch) =>{
  return{
    onHandleChangeCity: (event) =>dispatch(handleChangeCity(event.target.value)),
    fetchWeather: () => dispatch(fetchWeather())
  }
}

export class Main extends Component {
  state = {
    city: this.props.user.city,
    currentTemp: '',
    currentWeath: '',
    currentIcon: '',
    currentCity: '',
    user:this.props.user,
    err:false
  }
    // fetch api 
  componentDidMount(){
    this.props.fetchWeather()
  
  }
  // fetch api on submitt
  handleSubmit = (e) =>{
    e.preventDefault()
    this.props.fetchWeather()
    }


  render() {
    
    return (
      <div className='container'>
        <Home 
            city={this.props.currentCity}
            currentTemp={this.props.currentTemp}
            currentWeath={this.props.currentWeath}
            currentIcon={this.props.currentIcon}
            name={this.state.user.name}
            isPending={this.props.isPending}
            />

            <Search 
            handleSubmit={this.handleSubmit}
            handleChange={this.props.onHandleChangeCity}
            error={this.props.error}
            />
      </div>
      
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Main);
