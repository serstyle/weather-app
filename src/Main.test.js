import React from 'react';
import {shallow} from 'enzyme'
import {Main} from './Main';

let wrap

beforeEach(()=>{
  const fetchWeather = jest.fn()
  const props = {
    user:{
      city:'Paris',
      name:'fr'
    }
  }
  wrap = shallow(<Main {...props} fetchWeather={fetchWeather}/>)
})


it('should snapshot component', () => {
  expect(wrap).toMatchSnapshot()
})

it('test state', ()=>{
  expect(wrap.state().user.name).toEqual('fr')
  expect(wrap.state().city).not.toEqual('fr')
  expect(wrap.state().city).toEqual('Paris')
})

it('test handleSubmit',()=>{
  const submit = jest.spyOn(wrap.instance(), 'handleSubmit')
  submit({preventDefault: () => {}})
  expect(submit)
  .toHaveBeenCalledTimes(1)
  expect(wrap).toMatchSnapshot()
})