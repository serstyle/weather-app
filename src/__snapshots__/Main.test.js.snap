// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    fetchWeather={
      [MockFunction] {
        "calls": Array [
          Array [
            "Paris",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    user={
      Object {
        "city": "Paris",
        "name": "fr",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Home
          name="fr"
        />,
        <Search
          handleSubmit={[Function]}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "city": undefined,
          "currentIcon": undefined,
          "currentTemp": undefined,
          "currentWeath": undefined,
          "isPending": undefined,
          "name": "fr",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": undefined,
          "handleChange": undefined,
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Home
            name="fr"
          />,
          <Search
            handleSubmit={[Function]}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "city": undefined,
            "currentIcon": undefined,
            "currentTemp": undefined,
            "currentWeath": undefined,
            "isPending": undefined,
            "name": "fr",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": undefined,
            "handleChange": undefined,
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`test handleSubmit 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    fetchWeather={
      [MockFunction] {
        "calls": Array [
          Array [
            "Paris",
          ],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    user={
      Object {
        "city": "Paris",
        "name": "fr",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Home
          name="fr"
        />,
        <Search
          handleSubmit={[Function]}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "city": undefined,
          "currentIcon": undefined,
          "currentTemp": undefined,
          "currentWeath": undefined,
          "isPending": undefined,
          "name": "fr",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": undefined,
          "handleChange": undefined,
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Home
            name="fr"
          />,
          <Search
            handleSubmit={[Function]}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "city": undefined,
            "currentIcon": undefined,
            "currentTemp": undefined,
            "currentWeath": undefined,
            "isPending": undefined,
            "name": "fr",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": undefined,
            "handleChange": undefined,
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
